packages/kbn-test/src/failed_tests_reporter/github_api.ts

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import Url from 'url';
+
+import Axios, { AxiosRequestConfig } from 'axios';
+import parseLinkHeader from 'parse-link-header';
+import { ToolingLog } from '@kbn/dev-utils';
+
+const ISSUES_URL = '/repos/elastic/kibana/issues/';
+
+export interface GithubIssue {
+  html_url: string;
+  number: number;
+  title: string;
+  labels: unknown[];
+  body: string;
+}
+
+export class GithubApi {
+  private readonly x = Axios.create({
+    baseURL: 'https://api.github.com',
+    headers: {
+      Authentication: `token ${this.token}`,
+      'User-Agent': 'elastic/kibana#failed_test_reporter',
+    },
+  });
+
+  /**
+   * Create a GithubApi helper object, if token is undefined requests won't be
+   * sent, but will instead be logged.
+   */
+  constructor(private readonly log: ToolingLog, private readonly token: string | undefined) {}
+
+  async getKibanaIssues() {
+    const issues: GithubIssue[] = [];
+    const queue = [ISSUES_URL];
+
+    while (queue.length) {
+      const url = queue.shift()!;
+
+      const resp = await this.request<GithubIssue[]>({ method: 'GET', url }, []);
```

Comment:
This will only fetch open issues by default, you need to specify `?state=all` for open and closed issues. You can also speed this up quite a bit by filtering by the `failed-test` label we used before and using `per_page=100`.

If I remember correctly, paging through all of the github issues will take minutes if you don't filter by the label.